src/lib/npmCommand.ts

```diff
@@ -160,11 +156,73 @@ export class NpmModule {
   }
 
   public show(registry: string): NpmShowResults {
-    return NpmCommand.runNpmCmd(`show ${this.module}@${this.version}`, { registry, cliRoot: this.cliRoot });
+    const showCmd = NpmCommand.runNpmCmd(`show ${this.module}@${this.version} --json`, {
+      registry,
+      cliRoot: this.cliRoot,
+    });
+
+    // `npm show` doesn't return exit code 1 when it fails to get a specific package version
+    // If `stdout` is empty then no info was found in the registry.
+    if (showCmd.stdout === '') {
+      throw new SfdxError(`Failed to find ${this.module}@${this.version} in the registry`, 'NpmError');
+    }
+
+    try {
+      return JSON.parse(showCmd.stdout) as NpmShowResults;
+    } catch (error) {
+      throw new SfdxError(error, 'ShellParseError');
+    }
   }
 
   public pack(registry: string, options?: shelljs.ExecOptions): void {
-    NpmCommand.runNpmCmd(`pack ${this.module}@${this.version}`, { ...options, registry, cliRoot: this.cliRoot });
+    try {
+      NpmCommand.runNpmCmd(`pack ${this.module}@${this.version}`, {
+        ...options,
+        registry,
+        cliRoot: this.cliRoot,
+      });
+    } catch (err) {
+      const sfdxErr = SfdxError.wrap(err);
+      throw new SfdxError(`Failed to fetch tarball from the registry: \n${sfdxErr.message}`, 'NpmError');
+    }
     return;
   }
+
+  public async fetchTarbarll(registry: string, options?: shelljs.ExecOptions): Promise<void> {
+    await this.pollForAvailability(() => {
+      this.pack(registry, options);
+    });
+    this.pack(registry, options);
+  }
+
+  public async pollForAvailability(checkFn: () => void): Promise<void> {
+    const ux = await UX.create();
+    const isNonTTY = process.env.CI !== undefined || process.env.CIRCLECI !== undefined;
```

Comment:
is `process.stdout.isTTY` not reliable for CI environments?

Reply:
It should work too but haven't tried it, this command is executed by `plugin-rel-mgmt`.
`CI` and `CIRCLE` are built-in env vars in circle: https://circleci.com/docs/2.0/env-vars/#built-in-environment-variables